Monika After Story/game/script-islands-event.rpy

```diff
@@ -4,182 +4,899 @@
+            ParallaxDecal,
+            x=245,
+            y=-48,
+            z=1
+        )
+    )
+    _IslandsImgDataHolder(
+        "decal_tree",
+        partial_disp=functools.partial(
+            ParallaxDecal,
+            x=149,
+            y=-228,
+            z=3,
+            on_click="mas_island_cherry_blossom_tree"
+        )
+    )
+    GLITCH_FPS = (
+        "glitch/g_0.obj",
+        "glitch/g_1.obj",
+        "glitch/g_2.obj",
+        "glitch/g_3.obj",
+        "glitch/g_4.obj",
+        "glitch/g_5.obj",
+        "glitch/g_6.obj"
+    )
+    _IslandsImgDataHolder(
+        "decal_glitch",
+        fp_map={},
+        partial_disp=functools.partial(
+            ParallaxDecal,
+            x=260,
+            y=-44,
+            z=2,
+            on_click="mas_island_glitchedmess"
+        )
+    )
+    SHIMEJI_CHANCE = 100
+    _IslandsImgDataHolder(
+        "decal_shimeji",
+        fp_map={},
+        partial_disp=functools.partial(
+            ParallaxSprite,
+            Transform(renpy.display.image.ImageReference(("chibika", "smile")), zoom=0.4),
+            x=935,
+            y=325,
+            z=45,
+            on_click="mas_island_shimeji"
+        )
+    )
+    # BGs
+    _IslandsImgDataHolder(
+        "bg_def",
+        default_unlocked=True,
+        partial_disp=functools.partial(
+            ParallaxSprite,
+            x=0,
+            y=0,
+            z=15000,
+            min_zoom=1.1,
+            max_zoom=4.1,
+            on_click="mas_island_sky"
+        )
+    )
+    # Overlays
+    _IslandsImgDataHolder(
+        "overlay_rain",
+        default_unlocked=True,
+        fp_map={
+            "d": "overlays/rain/d.obj",
+            "n": "overlays/rain/n.obj"
+        },
+        partial_disp=functools.partial(
+            MASFilterWeatherDisplayable,
+            use_fb=True
+        )
+    )
+    _IslandsImgDataHolder(
+        "overlay_snow",
+        default_unlocked=True,
+        fp_map={
+            "d": "overlays/snow/d.obj",
+            "n": "overlays/snow/n.obj"
+        },
+        partial_disp=functools.partial(
+            MASFilterWeatherDisplayable,
+            use_fb=True
+        )
+    )
+
+
+    def _select_img(st, at, mfwm):
+        """
+        Selection function to use in Island-based images
+
+        IN:
+            st - renpy related
+            at - renpy related
+            mfwm - MASFilterWeatherMap for this island
+
+        RETURNS:
+            displayable data
         """
-        DynamicDisplayable for Island background images. This includes
-        special handling to return None if island images could not be
-        decoded.
+        # During winter we always return images with snow
+        # Nonideal, but we have to do this because of the tree
+        # FIXME: ideal solution would be split the images by seasons too
+        if store.mas_isWinter():
+            return mfwm.fw_get(store.mas_sprites.get_filter(), store.mas_weather_snow)
```

Comment:
```suggestion
            return mfwm.fw_get(store.mas_sprites.get_filter(), store.mas_weather_snow), None
```

`mas_fwm_select` returns a tuple of the image + None, which is intended for the return values for DynamicDisplayables. `fw_get` only returns a single value.  (unless you set the filter map's values to tuples, but I wouldnt expect that).